### PR TITLE
solutions: 437 - Path Sum III (Medium)

### PR DESCRIPTION
**Pull request by kesava-karri**
## Change Summary

Provide summary of changes with issue number if any.

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice. Regular contributors (with 10+ PRs) can skip this part.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [ ] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [ ] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [ ] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [ ] I've provided the full working solutions to the problems used in this tutorial.
- [ ] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [x] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [x] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [x] I've confirmed that comments are put above each line rather than writing on the same line.
- [x] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

### CODE REVIEW

File: solutions/0400-0499/0437-path-sum-iii-medium.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     * When the `remainingSum` reaches the node whose difference makes it zero leads to the path we need.
+     * So whenever such path exists we increment our answer
+     * Note: 
+     * 1. The start node doesn't need to be the root node
+     * 2. The end node doesn't have to be the leaf node
+     * The 3 paths for example 1 equating to targetSum are: [5, 3], [5, 2, 1], [-3, 11]
+     * @param root the TreeNode of binary tree
+     * @param targetSum to check if sum exists
+     * @return ans indicates the number of paths satisfying the given targetSum
+     * @see <a href="https://leetcode.com/problems/path-sum-ii/solutions/2615948/leetcode-the-hard-way-explained-line-by-line/">remainingSum example run thru by @wingkwong</a>
```
Comment (wingkwong) [inline solutions/0400-0499/0437-path-sum-iii-medium.md:75]: <a> tag won't work here in comment. just showing the link is fine.

Comment (wingkwong) [inline solutions/0400-0499/0437-path-sum-iii-medium.md:75]: image

Try to break the long sentences into different lines. The idea is to let the audience to read without scrolling horizontally.

Comment (kesava-karri) [reply solutions/0400-0499/0437-path-sum-iii-medium.md:75]: Acknowledged

Comment (kesava-karri) [reply solutions/0400-0499/0437-path-sum-iii-medium.md:75]: I agree with your feedback, let me change it and create new pr.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**